.github/workflows/_test.yaml

```diff
@@ -95,6 +96,48 @@ jobs:
           echo "matrix<<EOF" >> $GITHUB_OUTPUT
           echo "$BASE_MATRIX_ALL" >> $GITHUB_OUTPUT
           echo "EOF" >> $GITHUB_OUTPUT
+      - id: assign-e2e-runners
+        name: Assign runners for E2E tests
+        env:
+          IS_TRUSTED_BUILD: ${{ github.event_name == 'push' || github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository }}
+          ALL_RUNNERS: |-
+            {
+              "trusted_builds": {
+                "amd64": [
+                  "ubuntu-latest-kong"
+                ],
+                "arm64": [
+                  "ubuntu-latest-arm64-kong"
+                ]
+              },
+              "untrusted_builds": {
+                "amd64": [
+                  "ubuntu-latest"
+                ],
+                "arm64": []
+              }
+            }
+        run: |-
+          ALL_RUNNERS='${{ env.ALL_RUNNERS }}'
+          if [[ "${{ env.IS_TRUSTED_BUILD }}" == "true" ]]; then
+            ALL_RUNNERS=$(echo $ALL_RUNNERS | jq -r '.trusted_builds')
+          else
+            ALL_RUNNERS=$(echo $ALL_RUNNERS | jq -r '.untrusted_builds')
+          fi
+
+          for ARCH in amd64 arm64; do
+            COUNT=$(echo $ALL_RUNNERS | jq -r ".$ARCH | length")
+            if [[ $COUNT -eq 0 ]]; then
+              ALL_RUNNERS=$(echo $ALL_RUNNERS | jq -r ".$ARCH=\"\"")
+            else
+              ALL_RUNNERS=$(echo $ALL_RUNNERS | jq -r ".$ARCH=.$ARCH[0]")
+            fi
+          done
+
+          echo "final runners: $ALL_RUNNERS"
+          echo "runners<<EOF" >> $GITHUB_OUTPUT
+          echo "$ALL_RUNNERS" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
```

Comment:
Isn't that just:

```shell
ALL_RUNNERS='{"amd64": "ubuntu-latest", "arm64": "ubuntu-latest-arm64"}'
if [[ "${{ evv.IS_TRUSTED_BUILD}}" == "true" ]]; then
  ALL_RUNNERS='{"amd64": "ubuntu-latest-kong", "arm64": "ubuntu-latest-arm64-kong"}'
fi

echo "final runners: $ALL_RUNNERS"
echo "runners<<EOF" >> $GITHUB_OUTPUT
echo "$ALL_RUNNERS" >> $GITHUB_OUTPUT
echo "EOF" >> $GITHUB_OUTPUT
```

?

Then in KM we can simply override `IS_TRUSTED_BUILD` to always be true.